core/src/main/scala/kafka/server/AutoTopicCreationManager.scala

```diff
@@ -77,16 +79,24 @@ class DefaultAutoTopicCreationManager(
 
   private val inflightTopics = Collections.newSetFromMap(new ConcurrentHashMap[String, java.lang.Boolean]())
 
+  /**
+   * Initiate auto topic creation for the given topics.
+   *
+   * @param requestOpt defined when creating topics on behalf of the client. The goal here is to preserve
+   *                   original client principal for auditing, thus needing to wrap a plain CreateTopicsRequest
+   *                   inside Envelope to send to the controller when forwarding is enabled.
+   */
   override def createTopics(
     topics: Set[String],
-    controllerMutationQuota: ControllerMutationQuota
+    controllerMutationQuota: ControllerMutationQuota,
+    requestOpt: Option[RequestChannel.Request]
```

Comment:
I'm trying to come up with a better name. Maybe it's ok to be explicit about the fact that this should be a metadata request. If we change to `Option[RequestContext]`, then maybe we can just call it `metadataRequestContext` or something like that.